lib/main.dart

```diff
@@ -102,22 +109,28 @@ Future<void> main({AppConfig appConfig = schulCloudAppConfig}) async {
       yield EmptyStateLicense();
     });
 
-    logger.d('Running…');
-    runApp(
-      FutureBuilder<void>(
-        future: services.allReady(),
-        builder: (_, snapshot) {
-          if (!snapshot.hasData) {
-            return Container(
-              color: Colors.white,
-              alignment: Alignment.center,
-              child: CircularProgressIndicator(),
-            );
-          }
+    logger.d('Waiting for services.');
+    await services.allReady();
+
+    // Set demo banner based on current user.
+    // TODO(marcelgarus): dipose id stream.
+    services.storage.userIdString
+        .map((idString) => Id<User>(idString))
+        .listen((userId) {
+      userId.resolve().listen((user) {
+        // TODO(marcelgarus): Don't hardcode role id.
+        final isDemo =
+            user?.roleIds?.contains(Id<Role>('0000d186816abba584714d02')) ??
```

Comment:
This is the role ID of a demo student. We also need that of a demo teacher: 0000d186816abba584714d03
Source: https://github.com/schul-cloud/schulcloud-server/blob/a2e15cb617916589cabbe18c0deea8499ed61080/backup/setup/roles.json#L301-L342

Or (even better): Check transitively included roles for `demo` (0000d186816abba584714d00): https://github.com/schul-cloud/schulcloud-server/blob/a2e15cb617916589cabbe18c0deea8499ed61080/backup/setup/roles.json#L248-L278